api/src/components/coauthor/controller.ts

```diff
@@ -0,0 +1,204 @@
+import * as response from 'lib/response';
+import * as publicationService from 'publication/service';
+import * as coAuthorService from './service';
+import * as I from 'interface';
+
+export const create = async (
+    event: I.AuthenticatedAPIRequest<I.CreateCoAuthorRequestBody, undefined, I.CreateCoAuthorPathParams>
+): Promise<I.JSONResponse> => {
+    try {
+        const publication = await publicationService.get(event.pathParameters.id);
+
+        // Does the publication exist?
+        if (!publication) {
+            return response.json(404, {
+                message: 'This publication does not exist.'
+            });
+        }
+
+        // Is this user the author of the publication?
+        if (publication?.user.id !== event?.user.id) {
+            return response.json(403, {
+                message: 'You do not have the right permissions for this action.'
+            });
+        }
+
+        // Is the publication in draft?
+        if (publication.currentStatus === 'LIVE') {
```

Comment:
Because of the hidden status, maybe we check that it is only draft and not if it is live, i.e
```
if (publication.currenttatus !== 'DRAFT')
```

So publications with hidden status dont pass this check too.

Reply:
Good point, I think we should leave til we have `hidden` status.